### ISSUE TITLE
Idea: Sinkholes

### ISSUE THREAD

**Issue by BrekiTomasson** [13rac1/Minecraft-Water-Erosion#20]
First of all - love the mod! This kind of thing was always on my wishlist when playing Minecraft, the ability for landscapes to change over time based on things like water and wind erosion. Wind is obviously still on my wishlist ;)

Now, as for my idea: A lot of times, I see small lakes burrowing underground and creating a larger, subterranean, lake. Often, there's only a single `grass_block` above it, leading to what I feel like a fairly large, unsupported piece of land. My suggestion would be for this mod to periodically do checks that include checking whatever is above the water it is affecting, breaking "floating" flowers and grass and having a chance to degrade "floating" `grass_block`, `dirt`, or `podzol` above it. 

Basically; if it looks 1 block up from the water block and finds one of those three types blocks, and then 1 up from that and it finds `air`, then it should be able to degrade that block. By the time that block becomes `sand` or `gravel`, gravity should do the rest.

**Comment by 13rac1** [comment]
Thanks! :heart:

I've actually already implemented sinkholes in an unreleased branch: Water blocks looking downwards for something erodable, then air. Eroding starts if air is found, but the search stops if rock is found. The only problem is I need to deal with underwater plants. May have to start checking all waterlogged objects? Not sure how to handle it correctly yet. :thinking: Some screenshots

**Version 1.1.1/1.1.2 - No Sinkholes**
![2020-04-13_22 52 47-sinkhole-before](https://user-images.githubusercontent.com/195061/85492022-5d60e680-b589-11ea-8fc9-b39061cf357d.jpg)


**Feature branch with Sinkholes**
![2020-04-13_22 53 17-sinkhole-after](https://user-images.githubusercontent.com/195061/85492035-605bd700-b589-11ea-8962-9801f507c2ea.jpg)
